include/particle.tcc

```diff
@@ -312,6 +313,70 @@ bool mpm::Particle<Tdim, Tnphases>::compute_stress(unsigned phase) {
   return status;
 }
 
+// Compute pressure
+template <unsigned Tdim, unsigned Tnphases>
+bool mpm::Particle<Tdim, Tnphases>::compute_pressure(unsigned phase,
+                                                     double dt) {
+  bool status = true;
+  double pressure, dpressure;
+  double bulk_modulus, youngs_modulus, poisson_ratio, dvolumetric_strain;
+
+  // Strain rate for reduced integration
+  Eigen::VectorXd strain_rate_centroid =
+      cell_->compute_strain_rate_centroid(phase);
+
+  // Set dimension of strain rate
+  Eigen::Matrix<double, 6, 1> strain_rate;
+  strain_rate.setZero();
+  switch (Tdim) {
+    case (1): {
+      strain_rate(0) = strain_rate_centroid(0);
+      break;
+    }
+    case (2): {
+      strain_rate(0) = strain_rate_centroid(0);
+      strain_rate(1) = strain_rate_centroid(1);
+      strain_rate(3) = strain_rate_centroid(2);
+      break;
+    }
+    default: {
+      strain_rate = strain_rate_centroid;
+      break;
+    }
+  }
+
+  // Get dstrain
+  Eigen::Matrix<double, 6, 1> dstrain;
+  dstrain = strain_rate * dt;
+
+  try {
+    // Check if  material ptr is valid
+    if (material_ != nullptr) {
+
+      // Get material properties
+      youngs_modulus = material_->property("youngs_modulus");
```

Comment:
You might want to check if the material already has a bulk modulus, if not do this.

Reply:
I'd suggest doing this in the material itself rather than here, and just get the `bulk_modulus`

Reply:
That way we have to make another public variable, and not all materials have bulk modulus. Would you want to add virtual function to `material.h`?